docs/source/prgpatterns/lockstep.rst

```diff
@@ -45,16 +142,19 @@ Collective Loop
     // `frame` is a list which must be traversed collectively
     while( frame.isValid() )
     {
-        uint32_t const workerIdx = cupla::threadIdx( acc ).x;
-        using ParticleDomCfg = IdxConfig<
-            frameSize,
-            numWorker
-        >;
-        ForEachIdx< ParticleDomCfg > forEachParticle( workerIdx );
-        forEachParticle(
-           [&]( uint32_t const linearIdx, uint32_t const idx )
+        uint32_t const workerIdx = cupla::threadIdx(acc).x;
+        constexpr uint32_t frameSize = 256;
+        constexpr uint32_t numWorkers = 42;
+        auto forEachParticleSlotInFrame = lockstep::makeForEach<frameSize, numWorkers>(workerIdx);
+        forEachParticleSlotInFrame(
+           [&](lockstep::Idx const idx)
+           {
+               // independent work, idx can be used to access a context variable
+           }
+        forEachParticleSlotInFrame(
+           [&](uint32_t const linearIdx)
```

Comment:
Perhaps I did not express my idea clearly during the VC. Do we really need to make a distinction between the two cases? I feel it confuses much more than helps, and simple `auto` and unified example would be easier to use for vast majority of cases

Reply:
You can use `auto` but if you like to enforce a data type you can do it.
This gives the user the possibility to avoid implicit casts in the lambda function.